Added tsr conversion methods for JSON and YAML.

psigen opened the pull request:
This fixes the naming convention on TSR serialization to use `to_`/`from_` and adds convenience conversion methods for JSON and YAML.

mkoval commented:
There are several references to `serialize_dict` and `deserialize_dict` in `serialization.py`. Can you update those to use the new names?

It's also worth grepping the directory for their use to make sure you didn't miss anything.

psigen commented:
Ok, refactored other referenced in prpy to use `to/from_dict`.

I also moved the CBiRRT TSR serialize functions into `cbirrt.py`, because nothing in the universe should be using them.  Nothing in prpy should even know they ever existed.

@mkoval: can you test that this still works for you?

psigen commented:
@mkoval, does this address your concerns?

mkoval commented:
:+1:
